tests/components/elgato/test_sensor.py

```diff
@@ -47,6 +52,92 @@ async def test_battery_sensor(hass: HomeAssistant) -> None:
     assert entry.entity_category == EntityCategory.DIAGNOSTIC
     assert entry.options == {"sensor": {"suggested_display_precision": 0}}
 
+    # Battery voltage sensor
+    state = hass.states.get("sensor.frenck_battery_voltage")
+    assert state
+    assert state.state == "3860"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.VOLTAGE
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "Frenck Battery voltage"
+    assert state.attributes.get(ATTR_STATE_CLASS) is SensorStateClass.MEASUREMENT
+    assert (
+        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfElectricPotential.VOLT
+    )
+    assert not state.attributes.get(ATTR_ICON)
+
+    entry = entity_registry.async_get("sensor.frenck_battery_voltage")
+    assert entry
+    assert entry.unique_id == "GW24L1A02987_voltage"
+    assert entry.entity_category == EntityCategory.DIAGNOSTIC
+    assert entry.options == {
+        "sensor": {"suggested_display_precision": 2},
+        "sensor.private": {
+            "suggested_unit_of_measurement": UnitOfElectricPotential.VOLT
+        },
+    }
+
+    # Charging current sensor
+    state = hass.states.get("sensor.frenck_charging_current")
+    assert state
+    assert state.state == "3008"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.CURRENT
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "Frenck Charging current"
+    assert state.attributes.get(ATTR_STATE_CLASS) is SensorStateClass.MEASUREMENT
+    assert (
+        state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == UnitOfElectricCurrent.AMPERE
+    )
```

Comment:
@emontnemery This feels weird in testing: The state is 3008, but the unit is Ampere.

Is there a better way to handle this in tests?

Reply:
That's wrong?

Reply:
Maybe?

That is why I am asking. The implementation works correctly in the resulting UI, but it seems odd...

Reply:
I don't see what it is, but the state should definitely have the converted value according to the unit of measurement.